File: week3/atakantekatan/704. Binary Search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1
        while l <= r:
            if nums[l] == target:
                return l
            elif nums[r] == target:
                return r
            
            mid = (l+r) // 2
            
            if target < nums[mid]:
                r = mid - 1
            elif target > nums[mid]:
                l = mid + 1
            elif target == nums[mid]:
                return mid
        return -1

File: week3/Can-Yavuz/704-binary-search.cs
public class Solution {
    public int Search(int[] nums, int target) {
            int left = 0;
            int right = nums.Length -1;

            while(left <= right){
                int mid = left + (right - left) /2 ; 
                if(nums[mid] == target)
                return mid;
                else if(nums[mid] < target){
                    left = mid +1;
                }else{
                    right = mid -1;
                }
            } 
                return -1;
    }
}